File: java/v2/hello/src/main/java/xyz/hoper/hello/service/HelloWorldService.java
package xyz.hoper.hello.service;

import io.grpc.stub.StreamObserver;
import xyz.hoper.hello.annotation.GrpcService;
import xyz.hoper.protobuf.GreeterGrpc;
import xyz.hoper.protobuf.HelloReply;
import xyz.hoper.protobuf.HelloRequest;

import java.util.Date;

@GrpcService
public class HelloWorldService extends GreeterGrpc.GreeterImplBase {

    @Override
    public void sayHello(HelloRequest request, StreamObserver<HelloReply> responseObserver) {
        HelloReply.Builder helloReplyOrBuilder = HelloReply.newBuilder();
        helloReplyOrBuilder.setTime(new Date().getTime());
        responseObserver.onNext(helloReplyOrBuilder.build());
        responseObserver.onCompleted();
    }
}

File: server/static/kotlin/v2/hello/src/main/kotlin/xyz/hoper/hello/service/HelloWorldService.kt
package xyz.hoper.hello.service

import io.grpc.stub.StreamObserver
import xyz.hoper.grpc.annotation.GrpcService
import xyz.hoper.protobuf.GreeterGrpc.GreeterImplBase
import xyz.hoper.protobuf.HelloReply
import xyz.hoper.protobuf.HelloRequest
import java.util.*

@GrpcService
class HelloWorldService : GreeterImplBase() {
    override fun sayHello(request: HelloRequest, responseObserver: StreamObserver<HelloReply>) {
        val helloReplyOrBuilder = HelloReply.newBuilder()
        helloReplyOrBuilder.time = Date().time
        responseObserver.onNext(helloReplyOrBuilder.build())
        responseObserver.onCompleted()
    }
}
